src/modules/shared/forms/components/checkbox-array.component.ts

```diff
@@ -103,31 +108,47 @@ export class FormCheckboxArrayComponent implements OnInit, DoCheck {
         this.conditionalFormControl(item.conditional.id).updateValueAndValidity();
 
       }
-
+      
     });
-
+    
     this.cdr.detectChanges();
-
+    
   }
-
-
+  
+  
   isChecked(value: string): boolean {
     return this.fieldArrayValues.includes(value);
   }
-
+  
   onChanged(e: Event): void {
-
+    
     const event = e.target as HTMLInputElement;
     const valueIndex = (this.fieldArrayControl.value as string[]).indexOf(event.value);
-
+    
     if (event.checked && valueIndex === -1) {
+      
+      (this.isItemExclusive(event.value) || this.isExclusiveChecked()) && this.fieldArrayControl.clear(); 
+      
       this.fieldArrayControl.push(new FormControl(event.value));
+      
     }
-
+    
     if (!event.checked && valueIndex > -1) {
       this.fieldArrayControl.removeAt(valueIndex);
     }
-
+    
+  }
+  
+  private isItemExclusive(value: string): boolean {
+    return this.exclusiveItem === value;
+  }
+  
+  private isExclusiveChecked(): boolean {
+    return ( this.exclusiveItem !== undefined && (this.fieldArrayControl.value as string[]).includes(this.exclusiveItem) );
+  }
+  
+  private lookForAndSetExclusive() {
+    this.items?.map(item => item.exclusive && (this.exclusiveItem = item.value))

```

Comment:
I think `forEach` is better applied here, since you are not using the "transformed array" in map for anything.